src/display/api.js

```diff
@@ -3098,6 +3107,14 @@ class WorkerTransport {
     const refStr = ref.gen === 0 ? `${ref.num}R` : `${ref.num}R${ref.gen}`;
     return this.#pageRefCache.get(refStr) ?? null;
   }
+
+  switchToEditingMode(isEditing) {
```

Comment:
Sorry, but as mentioned in https://github.com/mozilla/pdf.js/pull/18134#pullrequestreview-2090293583 I still don't understand how it's safe and correct to essentially "mix" global state into the `getOperatorList` functionality on the worker-thread!?

Should we not instead include the `modifiedIds`-data into *each* `render`-call, such that it's actually constant and the result cannot change intermittently depending on when the `switchToEditingMode`-method happens to be invoked?
Hence, in order to change the `modifiedIds` we'd simply abort the current rendering (as usual) and trigger rendering again.

---

It seems to me that it'd should not be that difficult to include e.g. `modifiedIds` in the data returned by `getRenderingIntent`, and I could try to outline a way to do that if it'd be useful.

Reply:
My idea was just to avoid to always pass the same data in the worker.